downsampler_plugin/series_state.go

```
@@ -0,0 +1,98 @@
+//
+// ### Memory Safety:
+// - Only one message buffered per series (bounded memory usage)
+// - Messages are copied to prevent mutation issues
+// - Automatic cleanup on series reset or shutdown
+type SeriesState struct {
+	processor *algorithms.ProcessorWrapper
+	mutex     sync.Mutex
+
+	// Existing fields for late arrival detection and output tracking
+	lastProcessedTime time.Time
+	lastOutput        interface{}
```

golangci-lint (1.64.8)

field `lastOutput` is unused

(unused)